pkg/controllers/management/k3sbasedupgrade/template.go

```diff
@@ -36,130 +39,123 @@ var genericPlan = planv1.Plan{
 	Status: planv1.PlanStatus{},
 }
 
-func generateMasterPlan(version string, concurrency int, drain bool, upgradeImage, masterPlanName string) planv1.Plan {
-	masterPlan := genericPlan
-	masterPlan.Spec.Upgrade.Image = upgradeImage
-	masterPlan.Name = masterPlanName
-	masterPlan.Spec.Version = version
-	masterPlan.Spec.Concurrency = int64(concurrency)
+func generatePlan(plan planv1.Plan, name, version, image string, concurrency int, drain bool, selector *metav1.LabelSelector) planv1.Plan {
```

Comment:
It would make reviewing the diff much easier if you could move this function to the bottom of the file.

Reply:
Moved it to the bottom, didn't quite fix the diff. I can put it in a separate commit if you like.